solutions/CSES Subarray Sums II.mdx

```diff
@@ -0,0 +1,173 @@
+---
+id: cses-1661
+source: CSES
+title: Subarray Sums II
+author: Qi Wang
+---
+
+# Problem
+
+We are asked to find the number of subarrays that sum up to $x$ given the size of the array and its elements.
+
+<Spoiler title="Solution">
+
+We can have a map that keeps track of the prefix sums, we can check whether there are subarrays with prefixes with sum equal to $\texttt{prefixSum} - x$. This will ensure that we can remove a prefix from our current prefix to build a subarray with sum $x$.
```

Comment:
> we can check whether there are subarrays with prefixes with sum equal to

we can count the number of prefixes with sum equal to

Reply:
you never define what prefixSum means, or when you update the map.

Reply:
Okay, I changed prefixSum to prefixSum[i], maybe it will make more sense.

Reply:
Then you should say "At each index $i$"